internal/commands/app/create.go

```diff
@@ -65,7 +65,35 @@ func (cmd *CommandCreate) Handler(profile *cli.Profile, ui terminal.UI, clients
 		return err
 	}
 
-	// TODO(REALMC-8134): Implement dry-run for app create command
+	if cmd.inputs.DryRun {
+		if cmd.inputs.DataSource != "" {
+			clusters, err := clients.Atlas.ClustersByGroupID(groupID)
+			if err != nil {
+				return err
+			}
+			var clusterName string
+			for _, cluster := range clusters {
+				if cmd.inputs.DataSource == cluster.Name {
+					clusterName = cluster.Name
+					break
+				}
+			}
+			if clusterName == "" {
+				return errors.New("failed to find Atlas cluster")
+			}
+		}
+
+		headers := []string{"Info", "Details"}
+		rows := make([]map[string]interface{}, 0, 4)
+		rows = append(rows, map[string]interface{}{"Info": "Client App ID", "Details": "N/A"})
+		rows = append(rows, map[string]interface{}{"Info": "Realm Directory", "Details": dir})
+		rows = append(rows, map[string]interface{}{"Info": "Realm UI", "Details": "N/A"})
+		if cmd.inputs.DataSource != "" {
+			rows = append(rows, map[string]interface{}{"Info": "Data Source", "Details": cmd.inputs.DataSource})
+		}
+		ui.Print(terminal.NewTableLog("Successful dry run of app create", headers, rows...))
```

Comment:
Open to suggestions on the N/A's and the overall msg

Reply:
lets keep the dry run output simple/concise as best we can here, we can basically report on two things:
* we _would_ create an app (from scratch/from app <id>) at <dir>
* we _would_ link a data source (cluster name/data lake)

i think the one wrinkle is if/when we cant resolve the data source specified, we could just update that last message to indicate as much: "we would not link a data source because the Atlas cluster/data lake, 'my-data-source', cannot be found"

we can also have a follow up command here to try running the same command without --dry-run included

Reply:
also I wanted to ask: why are we "resolving data source" differently here than with the `list_commands` below?  is there a way to consolidate those two?

Reply:
discussed offline, and consolidated